core/src/main/java/com/yahoo/oak/UnionCodec.java

```diff
@@ -76,22 +82,23 @@ protected UnionCodec(int firstBitSizeLimit, int secondBitSizeLimit, int thirdBit
      * @param firstBitSizeLimit an upper limit on the size of the first parameter (exclusive)
      *                          if invalid calculate according to other two limits
      * @param secondBitSizeLimit an upper limit on the size of the second parameter (exclusive)
-     *                          if invalid calculate according to other two limits
+     * @param unionSizeInBits
      */
-    protected UnionCodec(int firstBitSizeLimit, int secondBitSizeLimit) {
+    protected UnionCodec(int firstBitSizeLimit, int secondBitSizeLimit, int unionSizeInBits) {
```

Comment:
Regarding using of the other constructor, putting zero in one of the parameters doesn't work well with `AUTO_CALCULATE_BIT_SIZE` and the following assert check. To rewrite the constructor for three parties would make it complex. We can discuss it.

I initiated third parameters in union for two with zero.

Reply:
I can fix it

Reply:
After the change all the test were failing. Unfortunately I do not have the time to debug. As this is not a critical change I reverted the change. This can be changed and debugged later.